recipe-server/client/control_new/components/recipes/DetailsActionBar.js

```diff
@@ -0,0 +1,150 @@
+    const recipe = getRecipe(state, recipeId, new Map());
+    const revisionId = getUrlParam(state, 'revisionId', latestRevisionId);
+
+    return {
+      isLatest: isLatestRevision(state, revisionId),
+      isLatestApproved: isLatestApprovedRevision(state, revisionId),
+      isPendingApproval: isRevisionPendingApproval(state, revisionId),
+      isApprovable: isApprovableRevision(state, revisionId),
+      recipe,
+      recipeId,
+      revisionId,
+    };
+  },
+  {
+    disableRecipe: recipeActions.disableRecipe,
+    enableRecipe: recipeActions.enableRecipe,
+    requestRevisionApproval: revisionActions.requestRevisionApproval,
+  },
+)
+@autobind
+export default class DetailsActionBar extends React.Component {
+  static propTypes = {
+    disableRecipe: PropTypes.func.isRequired,
+    enableRecipe: PropTypes.func.isRequired,
+    isApprovable: PropTypes.bool.isRequired,
+    isLatest: PropTypes.bool.isRequired,
+    isLatestApproved: PropTypes.bool.isRequired,
+    isPendingApproval: PropTypes.bool.isRequired,
+    recipe: PropTypes.object.isRequired,
+    recipeId: PropTypes.number.isRequired,
+    requestRevisionApproval: PropTypes.func.isRequired,
+    revisionId: PropTypes.string.isRequired,
+  };
+
+  handleDisableClick() {
+    const { disableRecipe, recipeId } = this.props;
+    Modal.confirm({
+      title: 'Are you sure you want to disable this recipe?',
+      onOk() {
+        disableRecipe(recipeId);
+      },
+    });
+  }
+
+  handlePublishClick() {
+    const { enableRecipe, recipeId } = this.props;
+    Modal.confirm({
+      title: 'Are you sure you want to publish this recipe?',
+      onOk() {
+        enableRecipe(recipeId);
+      },
+    });
+  }
+
+  handleRequestClick() {
+    const { requestRevisionApproval, revisionId } = this.props;
+    requestRevisionApproval(revisionId);
+  }
+
+  render() {
+    const {
+      isApprovable,
+      isLatest,
+      isLatestApproved,
+      isPendingApproval,
+      recipe,
+      recipeId,
+      revisionId,
+    } = this.props;
+
+    let cloneUrl = `/recipe/${recipeId}`;
+    if (revisionId) {
+      cloneUrl += `/rev/${revisionId}`;
+    }
+    cloneUrl += '/clone';
+
+    return (
+      <div className="details-action-bar clearfix">
+        <Link href={cloneUrl}>
+          <Button icon="swap" type="primary">Clone</Button>
+        </Link>
+
+        {
+          isLatest &&
+            <Link href={`/recipe/${recipeId}/edit`}>
+              <Button icon="edit" type="primary">Edit</Button>
+            </Link>
+        }
+
+        {
+          isApprovable &&
+            <Button icon="question-circle" type="primary" onClick={this.handleRequestClick}>
+              Request Approval
+            </Button>
+        }
+
+        {
+          isPendingApproval &&
+            <Link href={`/recipe/${recipeId}/approval_history`}>
+              <Button icon="message" type="primary">Approval Request</Button>
```

Comment:
Fair point re: animated icon, though I still think the treatment of the `pending approval` state needs a bit more prominence. As it stands, the only visual cue we're giving to the user that the recipe is pending approval is the `Pending Approval` text in the Timeline, and a 12x12 icon in a button.

I think we can do more to visually signify "hey this recipe is pending approval and this button can help you wrap it up." Maybe use an [Alert](https://ant.design/components/alert/) near the top of the content, and drop a button in there with some text?